thentos-core/src/Thentos/Transaction/TH.hs

```diff
@@ -26,41 +29,80 @@ import Thentos.Types (ThentosError, Extends)
 
 data TransactionType = TransQuery | TransUpdate
 
+-- | This function takes a database type and a list of user-defined transactions
+-- that may look like this:
+--
+-- > trans_foo :: (db `Extends` MyDB) => Bool -> ThentosQuery db Int
+--
+-- (where MyDB is the passed-in db type).
+-- and generates transactions of the form:
+-- 
+-- > (db `Extends` MyDB) => 
```

Comment:
Maybe `foo :: ...` so it's analogous to the one above, and it's more obvious how the functions are renamed?